### DISCUSSION TITLE
Explain package manager supported versions

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#24813] [Suggest an Idea]
### Describe the proposed change(s).

We should explain which versions of each package manager we support. There are multiple ways we could do this.

## Idea 1: put supported versions in code

We could maybe create a `const` like `supportedVersionsOfManager` and put the supported major versions in there:

`const supportedVersionsOfManager = [6, 7, 8]`

And then we auto-generate a "supported versions of package manager" table in the `modules/manager/manager_name` docs, like this:

| Package manager name | Supported versions |
| --- | --- |
| npm | 6, 7, 8 |

We might even create one big table for an overview of the versions and managers:

| Package manager name | Supported versions |
| --- | --- |
| npm | 6, 7, 8 |
| Yarn | 1, 2, 3 |
| Gradle | 6, 7 |
| Composer | ... |
| Cargo | ... |
| Bundler | ... |

The big drawback of this idea is that it's hard to explain that some versions have known problems when you use them with Renovate. For example how do you put in the code: "supports npm 8 except resolutions"?

## Idea 2: put supported version in `readme.md` and create a combined view table

The simplest thing would be:

- Put supported version info in each package manager's `readme.md`
- Have a combined view table that we update manually each time we add/remove/edit a supported version in the package `readme.md`

It's easier to document versions going forwards compared to trying to work out which old versions we're still compatible with.

We should distinguish between "known not working" older versions: doesn't work, we don't plan to support it, versus ones we simply don't know.

**Comment by rarkins** [comment]
In general I prefer this type of thing as code, but that's because it gives you the potential to "enforce" and link the code and comments. In this case, having an array of supported versions wouldn't be used anywhere, so is no more likely to stay up to date than if it's in a README. The main benefit would be the ability to more easily convert them all into a single table somewhere, as you point out.

**Comment by HonkingGoose** [comment]
> In this case, having an array of supported versions wouldn't be used anywhere, so is no more likely to stay up to date than if it's in a README.

What if we start using the array of supported versions in the code?

That way the version is always current, because people will open an issue to complain if their newer package manager version doesn't work with Renovate anymore. Then the programmers change the code, and bump the supported version number in the code, which then also changed the docs automatically.

**Comment by rarkins** [comment]
I feel like that's making the user experience worse in exchange for making docs better
